### PR TITLE
Display generic CV fetched from /generatepdf endpoint

### PR DESCRIPTION
**Pull request by fionaphillips**
Resolves [CG-46](https://unosquarebelfast.atlassian.net/secure/RapidBoard.jspa?rapidView=12&projectKey=CG&modal=detail&selectedIssue=CG-46)

Dependencies
- [x] https://github.com/UnosquareBelfast/CVGenerator-frontend/pull/20
- [x] https://github.com/UnosquareBelfast/CVGenerator-frontend/pull/21

fetchingCV

### CODE REVIEW

File: src/components/CVModal/container.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { fetchCV } from 'Services';
+import { getEmployeeCV } from 'Reducers/selected-options';
+
+const CVModalContainer = Wrapped =>
+  class extends React.Component {
+    static propTypes = {
+      employeeCV: PT.shape({ id: PT.number, label: PT.string }).isRequired,
+    };
+
+    state = { cv: '' };
```
Comment (girwin90) [inline src/components/CVModal/container.js]: Can this be null? Never been a big fan of empty strings personally but if needs be then fair enough.

File: src/components/CVModal/container.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    };
+
+    render() {
+      const { employeeCV, modalIsOpen, cv } = this.state;
+
+      return (
+        <Wrapped
+          employeeCV={employeeCV}
+          handleCancelClick={this.handleCancelModal}
+          modalIsOpen={modalIsOpen}
```
Comment (girwin90) [inline src/components/CVModal/container.js]: A small thing but I'd change this to be isModalOpen that's more a conventional way I see variable names that are boolean.

File: src/components/CVModal/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { PropTypes as PT } from 'prop-types';
+import { CancelModalButton, DownloadModalButton } from 'Components';
+import { Modal } from 'Common';
+import container from './container';
+
+export const CVModal = ({ employeeCV, handleCancelClick, modalIsOpen, cvData }) => {
+  if (employeeCV && employeeCV.length >= 1) {
```
Comment (girwin90) [inline src/components/CVModal/index.js]: I think this component should be pure with not much logic in it. I feel this component shouldn't be rendered if the if block exists in the container if it makes sense?

File: src/services/generate-cv-service.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { generatepdfEndpoint } from './config';
+
+export const fetchCV = () =>
+  fetch(generatepdfEndpoint, {
```
Comment (girwin90) [inline src/services/generate-cv-service.js:4]: I'd change the filename to be generateCVService.js and if firing multiple requests off maybe set up the headers in a utility sendAPIRequest func or something along those lines so you don't set the headers for each service func.

**Review by girwin90 — CHANGES_REQUESTED** [review]

File: src/components/CVModal/index.js (hunk 3 above)
Comment (fionaphillips) [reply src/components/CVModal/index.js]: moving the if into the container

File: src/services/generate-cv-service.js (hunk 4 above)
Comment (fionaphillips) [reply src/services/generate-cv-service.js:4]: @girwin90 is camel case the preferred convention for filenames here? I have used kebab  case throughout the project so if so I will make the change everywhere.

File: src/components/CVModal/container.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { fetchCV } from 'Services';
+import { getEmployeeCV } from 'Reducers/selected-options';
+
+const CVModalContainer = Wrapped =>
+  class extends React.Component {
+    static propTypes = {
+      employeeCV: PT.shape({ id: PT.number, label: PT.string }).isRequired,
+    };
+
+    state = { cv: null };
```
Comment (girwin90) [inline src/components/CVModal/container.js:14]: What data does CV contain?

File: src/components/CVModal/container.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    componentDidMount() {
+      fetchCV().then(data => this.setState({ isModalOpen: false, cv: data }));
```
Comment (girwin90) [inline src/components/CVModal/container.js:17]: As commented above, not a big fan of cv as a variable. Can we more detailed? I can't think of the top of my head though.....hmmm.

File: src/components/CVModal/container.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    componentDidUpdate(prevProps) {
+      const { employeeCV } = this.props;
+      if (prevProps.employeeCV !== employeeCV) {
+        this.setState({ employeeCV, isModalOpen: true });
+      }
+    }
+
+    handleCancelModal = () => {
```
Comment (girwin90) [inline src/components/CVModal/container.js:27]: `handleCancelModal = () => this.setState({ isModalOpen: false });`

You can write it up as above. If you're basically returning something straight away. It's a bit cleaner to write it like this. I know you've done it in other places. It's just to make it consistent across the app.

File: src/components/CVModal/container.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      return this.setState({ isModalOpen: false });
+    };
+
+    render() {
+      const { employeeCV, isModalOpen, cv } = this.state;
+
+      if (employeeCV && employeeCV.length >= 1) {
```
Comment (girwin90) [inline src/components/CVModal/container.js:34]: Can this be `employeeCV && employeeCV.length > 0` ?

File: src/components/CVModal/index.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+export const CVModal = ({ handleCancelClick, isModalOpen, cvData }) => {
+  return (
+    <Modal isOpen={isModalOpen} title="Preview Document">
+      <div className="modal-content">
+        <div className="cv-content">
+          <object data={cvData} title="CV Data" />
+        </div>
+        <div className="button-bar">
+          <CancelModalButton className="cancel-button" handleClick={handleCancelClick} />
```
Comment (girwin90) [inline src/components/CVModal/index.js:15]: Do you need class names? Are you not using styled components?

File: src/components/common/Modal/styled.js [hunk 10]
```diff
@@ -9,30 +9,47 @@ export const StyledModal = styled(ReactModalAdapter)`
...
-    border: ${({ theme }) => theme.borders.borderModal};
     background: #fff;
-    overflow: auto;
     outline: none;
-    padding: 20px;
+    border-radius: 10px;
+    height: 90%;
+    margin-bottom: 50px;
+
+    .modal-content {
```
Comment (girwin90) [inline src/components/common/Modal/styled.js:23]: You shouldn't in an ideal world use class names etc when using styled components. They should be broken down into small components that can be reused etc. Might be a bit of work now to redo that bit, if you have time have a go.

File: src/services/generate-cv-service.js (hunk 4 above)
Comment (girwin90) [reply src/services/generate-cv-service.js:4]: I usually stick to camelCase myself, most projects I work on do also.

**Review by girwin90 — CHANGES_REQUESTED** [review]
Some comments but looking good.

**Comment by fionaphillips** [comment]
Closing in favour of https://github.com/UnosquareBelfast/CVGenerator-frontend/pull/25

File: src/components/CVModal/container.js (hunk 5 above)
Comment (fionaphillips) [reply src/components/CVModal/container.js:14]: the URL pointing to the CV. I have changed the var name to cvURL

File: src/components/CVModal/container.js (hunk 6 above)
Comment (fionaphillips) [reply src/components/CVModal/container.js:17]: I have changed the var name to cvURL

File: src/components/common/Modal/styled.js (hunk 10 above)
Comment (fionaphillips) [reply src/components/common/Modal/styled.js:23]: will break the CVModal component into 2 - content and button bar

**Pull request closed without merging**